slashing-protection/src/test/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidatorTest.java

```diff
@@ -23,51 +26,76 @@
 import org.jdbi.v3.core.Jdbi;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(DatabaseSetupExtension.class)
+@ExtendWith(MockitoExtension.class)
 class GenesisValidatorRootValidatorTest {
+  @Spy private MetadataDao metadataDao = new MetadataDao();
+
   @Test
   void checkGenesisValidatorReturnTrueForNewGVR(final Jdbi jdbi) {
     final GenesisValidatorRootValidator gvrValidator =
-        new GenesisValidatorRootValidator(jdbi, new MetadataDao());
-    assertThat(
-            gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(Bytes32.leftPad(Bytes.of(3))))
+        new GenesisValidatorRootValidator(jdbi, metadataDao);
+    final Bytes32 genesisValidatorsRoot = Bytes32.leftPad(Bytes.of(3));
+    assertThat(gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(genesisValidatorsRoot))
+        .isTrue();
+    assertThat(gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(genesisValidatorsRoot))
         .isTrue();
+
+    // the database interaction happens for the first call only, cached gvr is used for subsequent
```

Comment:
The comments help but do we need to mix testing the cache into all of these unit tests?

Reply:
Agreed. Created specific unit test case to verify cached gvr usage.